pkg/drivers/vfkit/vfkit.go

```diff
@@ -153,6 +153,32 @@ func readPidfile(pidfile string) (int, error) {
 	return pid, nil
 }
 
+func signalPidfile(pidfile string, sig syscall.Signal) error {
+	pid, err := readPidfile(pidfile)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+		// Already stopped.
+		os.Remove(pidfile)
+		return nil
+	}
+	process, err := os.FindProcess(pid)
+	if err != nil {
+		return err
+	}
+	log.Infof("Sending signal %q to pid %v", sig, pid)
+	if err := process.Signal(sig); err != nil {
+		if err != os.ErrProcessDone {
```

Comment:
It may be more portable to use process.Kill() instead of process.Signal(SIGKILL).

Reply:
NB: one issue with `Kill` / `SIGKILL` is that the process does not have the opportunity to catch the signal to do some cleanup on its own before exiting, see https://github.com/containers/gvisor-tap-vsock/issues/485 for an example of this, but vfkit could have exactly the same problem.
This ties back to your comment in https://github.com/crc-org/vfkit/issues/278, vfkit client library should provide code which does the right thing.

Reply:
Yes, kill should be the last thing we try if stop did not work after some retries. Minikube use kill only when deleting a cluster, but maybe it should be better to try to stop gracefully and fallback to kill if there is no other way.